### PR TITLE
Improved maintainer experience

### PR DESCRIPTION
**Pull request by ulrikpedersen**
A few steps to tidy up and make life a little easier for the next developer/maintainer:

* Test mode when testing dls-release.py:  environment variables DLSBUILD_ROOT_DIR and/or DLSBUILD_WIN_ROOT_DIR can be used to override the default DLS queue location.
  * README.md updated with instructions/example
* Removing `from pkg_resources import require; require("blahblah")` from individual scripts as dependencies are defined in setuptools (i.e. setup.py)
  * This is a better practice as dependencies (and versions) are defined in a single central location as opposed to scattered across all the source files.
  * For some reason (unknown to me) this breaks nosetests, unless running it through setuptools so .travis.yml has been updated to do that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,12 +16,34 @@ Setup a virtualenv in the root of the project and install the dependencies:
   pip install -r requirements.txt
   pip install sphinx nose
```
Comment (willrogers) [inline README.md]: Nose is in requirements.txt. Should sphinx be there or should nose be removed from line 17?

Comment (ulrikpedersen) [reply README.md]: Good point. I think sphinx should be in requirements.txt.
This led me to try building the sphinx docs using the instructions in the README.md - and it currently doesn't work!!! I'll see if I can fix that to work

**Comment by mfurseman** [comment]
Just to make sure I understand, does setup.py create the entry points to the applications in prefix/bin and modifies the path there to ensure the applications run against the `install_requires=[...]` libraries?

**Comment by ulrikpedersen** [comment]
> Just to make sure I understand, does setup.py create the entry points to the applications in prefix/bin and modifies the path there to ensure the applications run against the install_requires=[...] libraries?

Essentially, yes.

**Pull request merged**